src/test/clojure/clara/test_keyword_names.clj

```diff
@@ -0,0 +1,48 @@
+(ns clara.test-keyword-names
```

Comment:
I think we can test this in ClojureScript using a strategy similar to [this](https://github.com/cerner/clara-rules/blob/0.17.0/src/test/clojurescript/clara/test_rules.cljs#L97) which pulls its rules data from [here](https://github.com/cerner/clara-rules/blob/0.17.0/src/test/clojurescript/clara/test_rules_data.clj#L5).  Basically raw Clojure data can be passed to the ClojureScript compiler path to create sessions.  We might not be able to test the case of failing due to duplicate production names so easily in ClojureScript though since we'd have to validate that ClojureScript compilation fails.  Every way I can think of to do that right now would be hard enough that it likely wouldn't be worthwhile.

Reply:
Done.

Reply:
There seems to be some duplication of rule and query definitions that isn't necessary.  The Clojure version should be able to use the same data definitions of the rules, so we shouldn't need to have more than one definition of each rule/query, and I don't see any reason to need more than two (clj and cljs).  You mentioned having some oddities with Git, perhaps multiple iterations on this were combined in your branch?

The way record types are referenced in Clojure and ClojureScript code forms is different, but we can get around that by using a custom :fact-type-fn that, say, looks at a specific field on a map that is expected to be a keyword.  I don't see immediately see anything else that would prevent sharing the rule/query structures between the Clojure and ClojureScript tests.